### ISSUE TITLE
[doc/enhancement] explicit guidance on building CPU_ONLY/CUDA version of pytorch

### ISSUE THREAD

**Issue by CDLuminate** [pytorch/pytorch#641]
Caffe and TensorFlow supports both CPU_ONLY and CUDA mode.
The meaning of CPU_ONLY mode is to enable users to run pytorch on architectures except for AMD64 and PPC64EL (Seems that CUDA is only available on AMD64 and PPC64EL).

For example, the CPU_ONLY version of caffe is available for (not only) amd64, armel, i386, ppc64el and s390x,
while the CUDA version only supports amd64 and ppc64el.

https://buildd.debian.org/status/package.php?p=caffe
https://buildd.debian.org/status/package.php?p=caffe-contrib

**Comment by apaszke** [comment]
I think we don't plan to invest too much time in supporting many different architectures. The vast majority of the users is on x86 and this will remain our main focus. If someone is willing to spend their time supporting other archs, we'll happily accept PRs, but we have a lot of other priorities.

We also support CPU only and CUDA mode. You can use or build pytorch without any GPU. We also have CPU-only binaries (see the website). You can force building without CUDA with `NO_CUDA=1` env variable, as I pointed out in another PR (it's broken because of THPP at the moment, but I'll fix it soon).

The only limitation for supporting different archs that I see is what our backends are limited to, and the fact that you need a C++11 compiler to build the extension. Apart from this, I don't think we depend on x86 anywhere.

Can you please what kind of guidance would you like to see? More detailed instructions on how to build pytorch somewhere in the docs?

**Comment by CDLuminate** [comment]
As reported in  [1], Pytorch's backend library (libTH) is now healthy in portability.

I suggest that the `NO_CUDA=1` environment variable should be referred in README.md .

[1] https://github.com/torch/torch7/issues/762
